numba/core/analysis.py

```diff
@@ -396,7 +396,12 @@ def resolve_input_arg_const(input_arg_idx):
         # literal type, return the type itself so comparisons like `x == None`
         # still work as e.g. x = types.int64 will never be None/NoneType so
         # the branch can still be pruned
-        return getattr(input_arg_ty, 'literal_type', Unknown())
+        try:
+            return getattr(input_arg_ty, 'literal_type', Unknown())
+        except ValueError:
```

Comment:
I think we should outside-in this fix and put the try..except in the property for `literal_type` to raise `AttributeError`

Reply:
Good idea. I also have a patch somewhere that makes `typeof(dict)` at least mostly behave, but I've a feeling that opens up a lot of complications for dicts in globals.